beacon-chain/state/stategen/replay_test.go

```diff
@@ -697,17 +697,17 @@ func TestLoadFinalizedBlocks(t *testing.T) {
 	gRoot, err := gBlock.Block.HashTreeRoot()
 	require.NoError(t, err)
 	util.SaveBlock(t, ctx, beaconDB, gBlock)
-	require.NoError(t, beaconDB.SaveGenesisBlockRoot(ctx, [32]byte{}))
+	require.NoError(t, beaconDB.SaveGenesisBlockRoot(ctx, gRoot))
 	roots, _, err := tree1(t, beaconDB, gRoot[:])
 	require.NoError(t, err)
 
-	filteredBlocks, err := s.loadFinalizedBlocks(ctx, 0, 8)
+	filteredBlocks, err := s.loadFinalizedBlocks(ctx, 0, 9)
 	require.NoError(t, err)
-	require.Equal(t, 0, len(filteredBlocks))
+	require.Equal(t, 1, len(filteredBlocks))
 	require.NoError(t, beaconDB.SaveStateSummary(ctx, &ethpb.StateSummary{Root: roots[8][:]}))
 
 	require.NoError(t, s.beaconDB.SaveFinalizedCheckpoint(ctx, &ethpb.Checkpoint{Root: roots[8][:]}))
-	filteredBlocks, err = s.loadFinalizedBlocks(ctx, 0, 8)
+	filteredBlocks, err = s.loadFinalizedBlocks(ctx, 0, 9)
 	require.NoError(t, err)
-	require.Equal(t, 10, len(filteredBlocks))
+	require.Equal(t, 7, len(filteredBlocks))
```

Comment:
This also manifests the purpose of the new algorithm. The tree constructed in `tree1` has orphaned blocks, but they were expected to be finalized.